catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/AbstractCswSource.java

```diff
@@ -1645,6 +1686,237 @@ public void setMetacardTypes(List<MetacardType> types) {
     }
 
     /**
+     * If a source is flagged as registry type, use the following methods for CRUD
```

Comment:
this is the exact type of logic that should only apply to the Registry specific implementation - not all CswSource implementations.